src/dotnet/Telemetry/ExternalTelemetryProperties.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security;
+using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.Win32;
+using RuntimeEnvironment = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment;
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+
+namespace Microsoft.DotNet.Cli.Telemetry
+{
+    // Some properties we need for telemetry, that don't yet have suitable
+    // public API
+    internal static class ExternalTelemetryProperties
+    {
+        /// <summary>
+        /// For Windows, returns the OS installation type, eg. "Nano Server", "Server Core", "Server", or "Client".
+        /// For Unix, or on error, currently returns empty string.
+        /// </summary>
+        internal static string GetInstallationType()
+        {
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
+                return "";
+
+            const string Key = @"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion";
+            const string ValueName = @"InstallationType";
+
+            try
+            {
+                return (string)Registry.GetValue(Key, ValueName, defaultValue: "");
+            }
+            // Catch everything: this is for telemetry only.
+            catch (Exception e)
+            {
+                Debug.Assert(e is ArgumentException | e is SecurityException | e is InvalidCastException);
+                return "";
+            }
+        }
+
+        [DllImport("kernel32.dll", ExactSpelling = true, SetLastError = false)]
+        private static extern bool GetProductInfo(uint dwOSMajorVersion, uint dwOSMinorVersion, uint dwSpMajorVersion, uint dwSpMinorVersion, out uint pdwReturnedProductType);
+
+        /// <summary>
+        /// For Windows, returns the product type, loosely the SKU, as encoded by GetProductInfo().
+        /// For example, Enterprise is "4" (0x4) and Professional is "48" (0x30)
+        /// See https://msdn.microsoft.com/en-us/library/windows/desktop/ms724358(v=vs.85).aspx for the full list.
+        /// We're not attempting to decode the value on the client side as new Windows releases may add new values.
+        /// For Unix, or on error, returns an empty string.
+        /// </summary>
+        internal static string GetProductType()
+        {
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
+                return "";
+
+            try
+            {
+                if (GetProductInfo((uint)Environment.OSVersion.Version.Major, (uint)Environment.OSVersion.Version.Minor, 0, 0, out uint productType))
+                {
+                    return productType.ToString("D");
```

Comment:
Pass CultureInfo.InvariantCulture here.  I don't think the "D" format actually varies by culture (does it?) but better not to have to think about the risk.

Reply:
In a quick look at the code it apparently doesn't (at least for uint) but I'll add it.